src/katsdpcontroller/generator.py

```diff
@@ -1513,10 +1489,140 @@ def get_baseline_index(a1, a2):
         )
         xbgpu.static_gauges["xbgpu_expected_engines"] = 1.0
 
-    _add_task_sensors(g, streams, task_names)
     return xbgpu_group
 
 
+def _make_vgpu(
+    g: networkx.MultiDiGraph,
+    configuration: Configuration,
+    streams: Iterable[product_config.GpucbfTiedArrayResampledVoltageStream],
+) -> scheduler.LogicalNode:
+    # Ensure that streams is a sequence, not just an iterable.
+    streams = sorted(streams, key=lambda stream: stream.name)
+
+    n_engines = 1
+    n_substreams = 1
+    base_name = streams[0].name
+    vgpu_group = LogicalGroup(f"vgpu.{base_name}")
+    g.add_node(vgpu_group)
+
+    for stream in streams:
+        stream_sensors: List[Sensor] = [
+            # The timestamps are simply ADC sample counts
+            Sensor(
+                float,
+                f"{stream.name}.scale-factor-timestamp",
+                "Factor by which to divide instrument timestamps to convert to seconds",
+                "Hz",
+                default=stream.adc_sample_rate,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                float,
+                f"{stream.name}.sync-time",
+                "The time at which the digitisers were synchronised. Seconds since the Unix Epoch.",
+                "s",
+                # TODO: Make grandparent property for TARVstream
+                default=sync_time,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                float,
+                f"{stream.name}.bandwidth",
+                "The analogue bandwidth of the digitised band",
+                "Hz",
+                default=acv.bandwidth,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                float,
+                f"{stream.name}.pass-bandwidth",
+                "Bandwidth over which the CBF produces a flat response",
+                "Hz",
+                default=acv.pass_bandwidth,
```

Comment:
The pass-bandwidth of the tarv stream is the acv pass_bandwidth times the `--passband` parameter to vgpu. I'd suggest adding a constant for this to defaults.py (something like VGPU_PASSBAND) set to 0.9, which will then also be used when launching vgpu.